iOSDFULibrary/Classes/Utilities/Streams/DFUStreamZip.swift

```diff
@@ -80,17 +80,18 @@ internal class DFUStreamZip : DFUStream {
     }
     
     /**
-     Initializes the stream with URL to the ZIP file.
+     Initializes the stream with given data of a ZIP file.
      
-     - parameter urlToZipFile: URL to the ZIP file with firmware files and manifest.json file containing metadata.
+     - parameter zipFile: Content of the ZIP file with firmware files and manifest.json file containing metadata.
+     - parameter type:    The type of the firmware to use
      
      - throws: DFUStreamZipError when manifest file was not found or contained an error
      
      - returns: the stream
      */
-    convenience init(urlToZipFile: URL) throws {
-        let allTypes = FIRMWARE_TYPE_SOFTDEVICE | FIRMWARE_TYPE_BOOTLOADER | FIRMWARE_TYPE_APPLICATION
-        try self.init(urlToZipFile: urlToZipFile, type: allTypes)
+    convenience init(zipFile: Data, type: DFUFirmwareType) throws {
+        let url = try ZipArchive.createTemporaryFile(zipFile)
+        try self.init(urlToZipFile: url, type: type)
```

Comment:
Are we deleting the temp file after we're done with it here?

Reply:
Nope... but also we weren't deleting the temp files after unzipping before. But as this is a temporary folder, shouldn't it be cleared by the system automatically? When app is closed, for example?

Reply:
Sure, the system will eventually decide to clear it at some point, just thinking that if for some reason the user selects a wrong zip file of a big size, we don't want to keep a big copy of the file since it's temporary and for our own use case, as per apple's [docs](https://developer.apple.com/documentation/foundation/filemanager/1407693-url):

> It is recommended that you remove any temporary directories that are created after they're no longer needed.

Again, not a breaker and can be ignored for now :)